resource/resource_registry.go

```diff
@@ -74,7 +75,37 @@ type DependencyNotReadyError struct {
 }
 
 func (e *DependencyNotReadyError) Error() string {
-	return fmt.Sprintf("dependency %q is not ready yet; reason=%q", e.Name, e.Reason)
+	return fmt.Sprintf("dependency %q is not ready yet; reason=%s", e.Name, e.Reason)
+}
+
+func (e *DependencyNotReadyError) PrettyPrint() string {
```

Comment:
Nice! I like the indented style with `-`. How come we're putting this in a new, uncalled `PrettyPrint` method instead of just modifying the `Error` method?

Reply:
There's a claim that the robot logs in the UI will reverse the order because it displays newest logs on top.
There's also some aversion to having logs include formatting. That logs should be structured and log viewers should do formatting.

I'm not sure if the former is correct for when a single log call has newlines. I philosophically agree with the latter, but do think there's some room as a startup to do things the wrong but expedient way. Particularly when undoing the change would have negligible impact. Ultimately I don't want to go behind anyways back to sneak that in.

That said, I do plan on using this when developing locally with a robot.

Reply:
> I'm not sure if the former is correct for when a single log call has newlines.

Yeah I'd expect the log to print correctly even if the net-logged logs are being output in "reverse".

> That logs should be structured and log viewers should do formatting.

That sounds correct to me, too.

Using this as a local debug tool to debug-print complex dependency errors seems good. Could you add an export comment above this method to that end?